calour/sorting.py

```diff
@@ -132,8 +133,7 @@ def cluster_data(exp: Experiment, transform=None, axis=1,
         data = exp.get_data(sparse=False)
     else:
         logger.debug('transforming data using %r' % transform)
-        newexp = deepcopy(exp)
-        data = transform(newexp, **kwargs).get_data(sparse=False)
+        data = transform(exp, **kwargs, inplace=False).get_data(sparse=False)
```

Comment:
i think you need to add copy=True,
otherwise, if the experiment is already not sparse, will return the reference to the data instead of copy it (and then will change in place...)